pkg/controller/storagecluster/storagecluster_controller_test.go

```diff
@@ -304,11 +304,12 @@ func TestThrottleStorageDevices(t *testing.T) {
 							StorageClassName: &fakestorageClassName,
 						},
 					},
-					Portable: true,
+					Portable:   true,
+					DeviceType: "hdd",
```

Comment:
This seems to be a change to the test case that is not related to this code change. 🤔 

It explicitly configures the type "hdd" in the device set and correctly changes the expected result to Slow. But isn't that the test for the previous PR #944 ?

Reply:
to solve this error `	Error Trace:	storagecluster_controller_test.go:426
        	Error:      	Received unexpected error:
        	            	could not get infrastructure details to determine cloud platform: infrastructures.config.openshift.io "cluster" not found
`

Reply:
and yes for this we should have modify the creation of cluster with platform adding new test cases for this addidtion.

Reply:
@crombus - hmmm - wouldn't it be better / more natural to fix this problem by changing the cluster structure in the test so that the platform detection can not fail? It does not seem to be conceptually correct to fix it by avoiding the call in the first place by adding a configuration we know will be checked first... 

Can the platform detection fail in the real world? (not mocked-up test code)

Reply:
yes, it can fail. And cration of cluster with platform is the correct fix